File: Java/app/src/main/java/com/zivapp/notes/adapters/AdapterContacts.java
package com.zivapp.notes.adapters;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;

import androidx.annotation.NonNull;
import androidx.databinding.DataBindingUtil;
import androidx.recyclerview.widget.RecyclerView;

import com.zivapp.notes.R;
import com.zivapp.notes.databinding.ItemContactsBinding;
import com.zivapp.notes.model.User;
import com.zivapp.notes.views.contacts.ContactsListActivity;
import com.zivapp.notes.views.contacts.SelectedUsersListener;

import java.util.ArrayList;

/**
 * Adapter used for the {@link ContactsListActivity}.
 */
public class AdapterContacts extends RecyclerView.Adapter<AdapterContacts.ItemViewHolder> {

    private static final String TAG = "AdapterContacts";

    private ArrayList<User> list;
    private SelectedUsersListener mSelectedUsersListener;

    public AdapterContacts(ArrayList<User> list, SelectedUsersListener selectedUsersListener) {
        this.list = list;
        mSelectedUsersListener = selectedUsersListener;
    }

    public void setNoteList(ArrayList<User> list) {
        this.list = list;
    }

    public static class ItemViewHolder extends RecyclerView.ViewHolder {

        private ItemContactsBinding binding;

        public ItemViewHolder(@NonNull ItemContactsBinding binding) {
            super(binding.getRoot());
            this.binding = binding;
        }
    }

    @NonNull
    @Override
    public ItemViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        LayoutInflater inflater = LayoutInflater.from(parent.getContext());
        ItemContactsBinding binding = DataBindingUtil.inflate(inflater,
                R.layout.item_contacts, parent, false);
        return new ItemViewHolder(binding);
    }

    @Override
    public void onBindViewHolder(@NonNull final ItemViewHolder holder, final int position) {
        final User user = list.get(position);
        holder.binding.setUser(user);

        final ImageView pickContact = holder.binding.imagePicked;
        if (user.isSelected()) {
            pickContact.setVisibility(View.VISIBLE);
        } else {
            pickContact.setVisibility(View.GONE);
        }

        // pick a contact(s)
        holder.binding.userCardView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if (user.isSelected()) {
                    pickContact.setVisibility(View.GONE);
                    user.setSelected(false);
                    if (getSelectedUsers().size() == 0) {
                        mSelectedUsersListener.onSelectedAction(false);
                    }
                } else {
                    pickContact.setVisibility(View.VISIBLE);
                    user.setSelected(true);
                    mSelectedUsersListener.onSelectedAction(true);
                }
            }
        });
    }

    public ArrayList<User> getSelectedUsers() {
        ArrayList<User> usersList = new ArrayList<>();
        for (User user : list) {
            if (user.isSelected()) {
                usersList.add(user);
            }
        }
        return usersList;
    }

    @Override
    public int getItemCount() {
        if (list != null) {
            return list.size();
        } else {
            return 0;
        }
    }
}

File: Kotlin/app/src/main/java/com/zivapp/countandnote/adapters/AdapterContacts.kt
package com.zivapp.countandnote.adapters

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import androidx.databinding.DataBindingUtil
import androidx.recyclerview.widget.RecyclerView
import com.zivapp.countandnote.R
import com.zivapp.countandnote.databinding.ItemContactsBinding
import com.zivapp.countandnote.model.User
import com.zivapp.countandnote.view.contacts.SelectedUsersListener
import java.util.*

class AdapterContacts(selectedUsersListener: SelectedUsersListener) :
    RecyclerView.Adapter<AdapterContacts.ItemViewHolder>() {

    private val TAG = "AdapterContacts"
    private var mSelectedUsersListener: SelectedUsersListener = selectedUsersListener

    var list: List<User> = listOf()
        set(value) {
            field = value
            notifyDataSetChanged()
        }

    class ItemViewHolder(binding: ItemContactsBinding)
        : RecyclerView.ViewHolder(binding.root) {
        val mBinding: ItemContactsBinding = binding
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ItemViewHolder {
        val inflater = LayoutInflater.from(parent.context)
        val binding: ItemContactsBinding = DataBindingUtil.inflate(
            inflater,
            R.layout.item_contacts, parent, false
        )
        return ItemViewHolder(binding)
    }

    override fun onBindViewHolder(holder: ItemViewHolder, position: Int) {
        val user: User = list[position]
        holder.mBinding.user = user

        val pickContact: ImageView = holder.mBinding.imagePicked
        if (user.selected) {
            pickContact.visibility = View.VISIBLE
        } else {
            pickContact.visibility = View.GONE
        }

        // pick a contact(s)
        holder.mBinding.userCardView.setOnClickListener {
            if (user.selected) {
                pickContact.visibility = View.GONE
                user.selected = false
                if (getSelectedUsers().size == 0) {
                    mSelectedUsersListener.onSelectedAction(false)
                }
            } else {
                pickContact.visibility = View.VISIBLE
                user.selected = true
                mSelectedUsersListener.onSelectedAction(true)
            }
        }
    }

    fun getSelectedUsers(): ArrayList<User> {
        val usersList = ArrayList<User>()
        for (user in list) {
            if (user.selected) {
                usersList.add(user)
            }
        }
        return usersList
    }

    override fun getItemCount(): Int {
        return list.size
    }
}